python/src/spark_rapids_ml/umap.py

```diff
@@ -538,19 +678,22 @@ def _construct_umap() -> CumlT:
 
             from .utils import cudf_to_cuml_array
 
-            embedding_np = np.array(self.embedding, dtype=np.float32)
-            raw_data_np = np.array(self.raw_data, dtype=np.float32)
+            embedding = self.embedding
+            raw_data = self.raw_data
+            if embedding.dtype != np.float32:
+                embedding = embedding.astype(np.float32)
+                raw_data = raw_data.astype(np.float32)
```

Comment:
Probably should log a warning that we're auto-converting the type (but only if it's not too chatty).

Reply:
At the moment I'm not supporting user-control over the "convert_dtype" param from cuml (determines whether the internal computations are float64); currently just defaulting to float32. I figured we could keep it like that for now and add float64 support in a future pr (and like Erik mentioned, maybe include a default conversion to float32 much earlier in the computation if desired).